Tasks/NugetPublisher/NuGetPublisher.ps1

```diff
@@ -65,6 +54,95 @@ if (-not $nuGetPath)
     throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
 }
 
+#Setup Nuget
+Write-Host "Creating Nuget Arguments"
+
+if($connectedServiceName)
+{
+    $serviceEndpoint = GetEndpointData $connectedServiceName
+
+    Write-Verbose "serverUrl = $($serviceEndpoint.Url)"
+    $nugetServer = $($serviceEndpoint.Url)
+
+    Write-Verbose "serverApiKey = $($serviceEndpoint.Authorization.Parameters.Password)"
+    $nugetServerKey = $($serviceEndpoint.Authorization.Parameters.Password)
+
+    Write-Host "Checking server url set"
+    if (!$nugetServer)
+    {
+        throw "Server must be set"
+    }
+
+    Write-Host "Checking server key set"
+    if (!$nugetServerKey)
+    {
+        throw "Server Key must be set, set the password on the generic service"
+    }
+}
+elseif($feedName)
+{
+    $nugetServer = $feedName
+
+    #check if nuget config exists
+    if(-not (Test-Path -Path $tempNuGetConfigPath))
+    {
+        # Create basic NuGet config file if it doesn't exist
+        [System.Xml.XmlDocument] $nuGetConfig = New-Object System.Xml.XmlDocument
+
+        $configurationSection = $nuGetConfig.CreateElement("configuration")
+        [void]$nuGetConfig.AppendChild($configurationSection)
+    }
+    else
+    {
+        $nuGetConfig = [xml](Get-Content $tempNuGetConfigPath)
+        $configurationSection = $nuGetConfig.configuration
+    }
+
+    $packageSourcesSection = $nuGetConfig.SelectSingleNode("configuration/packageSources")
+    if($packageSourcesSection -eq $null)
+    {
+        $packageSourcesSection = $nuGetConfig.CreateElement("packageSources")
+        [void]$configurationSection.AppendChild($packageSourcesSection)   
+    }
+    
+    #check if URL in packageSources
+    $nuGetSources = $nuGetConfig.SelectNodes("configuration/packageSources/add")
+    $sourceExists = $false
+
+    foreach($source in $nuGetSources)
```

Comment:
Are theses URLs? If so, do you need to trim trailing slashes here before performing the comparison?